src/compas_viewer/scene/scene.py

```diff
@@ -70,10 +72,29 @@ def __init__(self, name: str = "ViewerScene", context: str = "Viewer"):
 
         #  Primitive
         self.objects: list[ViewerSceneObject]
-
+        self._observers = []
         #  Selection
         self.instance_colors: dict[tuple[int, int, int], ViewerSceneObject] = {}
         self._instance_colors_generator = instance_colors_generator()
+        # Time Debounce
+        self._time = None
+        self.update_timer = QTimer()
+        self.update_timer.setSingleShot(True)
+        self.update_timer.timeout.connect(self.update_observers)
+        self.debounce_interval = 200
+
+    @property
+    def observers(self):
+        from compas_viewer import Viewer
+
+        self.viewer = Viewer()
+        self._observers.extend(
+            [
+                self.viewer.renderer,
+                self.viewer.ui.sidebar,
+            ]
+        )
```

Comment:
I think this will repetitively expand the `_observer` list. You need to either make it a `Set` or check if things has already been added to list